### ISSUE TITLE
Detekt-cli Scan report based on baseline is wrong

### ISSUE THREAD

**Issue by opluscustom** [detekt/detekt#4234]
## Expected Behavior

In my way ,I scan the base-code smells ,and apply the patch.diff in check workspace. And then generate report based on detekt-base.xml
By expected,the Report should report the smells in correct diff lines

## Observed Behavior

When the patch has new lines or delete lines,the diff reports is wrong,because the line numbers of smells in detekt-base.xml has already changed

## Steps to Reproduce
1. Scan to generate detekt-base.xml
2. modify the code by add or delete some lines
3. scan to generate report based on detekt-base.xml

## Context
This problem make incremental scan reports inaccurate

## Your Environment

* Version of detekt used:
* Version of Gradle used (if applicable):
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version:
* Link to your project (if it's a public repository):

**Comment by BraisGabin** [comment]
Sorry but could you show an example? I use the baseline a lot in my project and I modify the code around the issue and I don't need to recreate the baseline.

**Comment by opluscustom** [comment]
> Sorry but could you show an example? I use the baseline a lot in my project and I modify the code around the issue and I don't need to recreate the baseline.
It is not recreate baseline. when use cli like this:
 1. java -jar xxx.jar -i test.kt -b detekt-base.xml -cb ,you may have  some issues in base xml
 2. modify test.kt by add some new lines or detele some lines 
 3. java -jar xxx.jar -i test.kt -b detekt-base.xml -r xml:detekt-diff.xml. The diff xml will report the same issue in base xml ,because the line number of the issue has changed
 
 I am sorry that I can not provide the example because I am working on the cloud server.

**Comment by BraisGabin** [comment]
I created this file:
```kotlin
class Foo {
    private val a = 50

    fun b() {
        // TODO: implement
    }
}
```

It have 3 different issues: magic number, unused private and forbidden comment.

When I run `java -jar ~/Downloads/detekt-cli-1.18.1-all.jar -i Foo.kt -b detekt-base.xml -cb` I get this file in `detekt-base.xml`.

```xml
<?xml version="1.0" ?>
<SmellBaseline>
  <ManuallySuppressedIssues></ManuallySuppressedIssues>
  <CurrentIssues>
    <ID>ForbiddenComment:Foo.kt$Foo$// TODO: implement</ID>
    <ID>MagicNumber:Foo.kt$Foo$50</ID>
    <ID>UnusedPrivateMember:Foo.kt$Foo$private val a = 50</ID>
  </CurrentIssues>
</SmellBaseline>
```

Then, I add new lines everywhere in the `Foo.kt` file and I run this: `java -jar ~/Downloads/detekt-cli-1.18.1-all.jar -i Foo.kt -b detekt-base.xml -r xml:detekt-diff.xml` and the content on `detekt-diff.xml` is:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle version="4.3">
</checkstyle>
```

So I can't reproduce your issue.

**Comment by BraisGabin** [comment]
I created this file:
```kotlin
class Foo {
    private val a = 50

    fun b() {
        // TODO: implement
    }
}
```

It have 3 different issues: magic number, unused private and forbidden comment.

When I run `java -jar ~/Downloads/detekt-cli-1.18.1-all.jar -i Foo.kt -b detekt-base.xml -cb` I get this file in `detekt-base.xml`.

```xml
<?xml version="1.0" ?>
<SmellBaseline>
  <ManuallySuppressedIssues></ManuallySuppressedIssues>
  <CurrentIssues>
    <ID>ForbiddenComment:Foo.kt$Foo$// TODO: implement</ID>
    <ID>MagicNumber:Foo.kt$Foo$50</ID>
    <ID>UnusedPrivateMember:Foo.kt$Foo$private val a = 50</ID>
  </CurrentIssues>
</SmellBaseline>
```

Then, I add new lines everywhere in the `Foo.kt` file and I run this: `java -jar ~/Downloads/detekt-cli-1.18.1-all.jar -i Foo.kt -b detekt-base.xml -r xml:detekt-diff.xml` and the content on `detekt-diff.xml` is:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<checkstyle version="4.3">
</checkstyle>
```

So I can't reproduce your issue.

**Comment by opluscustom** [comment]
> ```
> <?xml version="1.0" ?>
> <SmellBaseline>
>   <ManuallySuppressedIssues></ManuallySuppressedIssues>
>   <CurrentIssues>
>     <ID>ForbiddenComment:Foo.kt$Foo$// TODO: implement</ID>
>     <ID>MagicNumber:Foo.kt$Foo$50</ID>
>     <ID>UnusedPrivateMember:Foo.kt$Foo$private val a = 50</ID>
>   </CurrentIssues>
> </SmellBaseline>
> ```
> 

Thank U for your Demo show，After comparison, I found the point of this problem:
I run with "java -jar ~/Downloads/detekt-cli-1.18.1-all.jar -c  detekt-formatting-1.18.1.jar -i xxx.kt -b detekt-base.xml -cb"
the problem is the rules in formatting-jar plugin will generate line number in detekt-base.xml,just like:
>     <ID>AnnotationSpacing:com.android.xxx.xxx.kt:18</ID>
>     <ID>MultiLineIfElse::com.android.xxx.xxx.kt:220</ID>

so this line number will affect the generation of differential reports.
the detekt-cli-1.18.1-all.jar&custom-xxx.jar have no such problem

**Comment by opluscustom** [comment]
So can we resolve this problem by update the detekt-formatting-1.18.1.jar?

**Comment by BraisGabin** [comment]
Thanks! I can reproduce it now. This is an issue in `detekt-formatting`.

**Comment by opluscustom** [comment]
When can this problem be fixed?
Or how can we avoid it

**Comment by BraisGabin** [comment]
I can think about two workarounds:

1. no use detekt-formatting and use ktlint directly for formatting issues. They have baseline functionality too and I imagine that it will not have this issue. I never used their baseline so I have no idea.
2. Fix all the formatting issues. That can be done automatically. The change will be huge but I never heard of someone having behaviour changes because of this.

About how this will be fixed: no ETA. But I encorage you to take a look a send a PR fixing it. It shouldn't be a difficult one. And if you need any help doing it don't hestiate to ask. You can use this issue or Slack.

**Comment by opluscustom** [comment]
OK，I try to fix it
